pkgs/ffigen/example/ffinative/lib/generated_bindings.dart

```diff
@@ -43,9 +45,27 @@ external ffi.Pointer<ffi.Float> divide(
 
 /// Divides 2 floats, returns a pointer to double.
 @ffi.Native<ffi.Pointer<ffi.Double> Function(ffi.Float, ffi.Float)>(
-    symbol: 'dividePrecision',
     assetId: 'package:ffinative_example/generated_bindings.dart')
 external ffi.Pointer<ffi.Double> dividePrecision(
   double a,
   double b,
 );
+
+@ffi.Array(10)
+@ffi.Native<ffi.Array<ffi.Int>>(
+    assetId: 'package:ffinative_example/generated_bindings.dart')
+external ffi.Array<ffi.Int> array;
+
+/// Version of the native C library
+@ffi.Native<ffi.Pointer<ffi.Char>>(
+    assetId: 'package:ffinative_example/generated_bindings.dart')
+external final ffi.Pointer<ffi.Char> library_version;
+const _SymbolAddresses addresses = _SymbolAddresses();
+
+class _SymbolAddresses {
+  const _SymbolAddresses();
+  ffi.Pointer<ffi.NativeFunction<ffi.Int Function(ffi.Int, ffi.Int)>> get sum =>
+      ffi.Native.addressOf(self.sum);
```

Comment:
Maybe add some more examples to ensure we have all corner cases.

* Non final field
* Array field
* Non-pinter field

Reply:
It looks like most of these are covered by the native globals test as well, but I have added a non-final + non-pointer field to the example to show how it works.